README.md

```diff
@@ -30,13 +30,56 @@ The concrete class that implements `IStatsDPublisher` is `StatsDPublisher`. The
 
 An example of a very simple StatsD publisher configuration, using the default values for most things.
 
+#### .NET Core
+
 ```csharp
+// Service registration
+services.AddStatsD("metrics_server.mycompany.com");
+services.AddSingleton<MyService>();
+
+// Consumption
+public class MyService
+{
+    public MyService(IStatsDPublisher statsPublisher)
+    {
+        // Use the publisher
+    }
+}
+
+```
+
+#### .NET 4.5.1
 
+```csharp
 var statsDConfig = new StatsDConfiguration { Host = "metrics_server.mycompany.com" };
 IStatsDPublisher statsDPublisher = new StatsDPublisher(statsDConfig);
 ```
 
-### IoC example
+### IoC Examples
+
+#### .NET Core
+
+An example of registering StatsD dependencies using `IServiceCollection`:
+
+```csharp
+services.AddStatsD(
+    (provider) =>
+    {
+        var options = provider.GetRequiredService<MyOptions>().StatsD;
+        var logger = provider.GetRequiredService<ILogger<StatsDConfiguration>>();
```

Comment:
Logger is not used in this example. Perahs should be in the `LogError(ex)` line?

Reply:
Good spot. It used to be, but then I took the code from the existing example and simplified it.